logs_sdk/lib/opentelemetry/sdk/logs/log_record.rb

```diff
@@ -103,6 +113,50 @@ def to_integer_nanoseconds(timestamp)
 
           (timestamp.to_r * 10**9).to_i
         end
+
+        def trim_attributes(attributes)
+          return if attributes.nil?
+
+          # truncate total attributes
+          truncate_attributes(attributes, @log_record_limits.attribute_count_limit)
+
+          # truncate attribute values
+          truncate_attribute_values(attributes, @log_record_limits.attribute_length_limit)
+
+          # validate attributes
+          validate_attributes(attributes)
+
+          nil
+        end
+
+        def truncate_attributes(attributes, attribute_limit)
+          excess = attributes.size - attribute_limit
+          excess.times { attributes.shift } if excess.positive?
+        end
+
+        def validate_attributes(attrs)
+          # Similar to Internal.valid_attributes?, but with different messages
```

Comment:
Can we update Internal.valid_attributes? to accept `log record` as a kind?

Reply:
We could! I wanted to minimize the impact on the tracing SDK and other gems, so I didn't reach toward refactoring it initially. Would you prefer a refactor over this duplication?

One way to update it could be to add a new `signal` attribute, refactor the message to remove `span`, and perhaps use `log_record.body` for `owner`: 
https://github.com/open-telemetry/opentelemetry-ruby/blob/98c629edbc6445305bed65c4f4306431d683b474/sdk/lib/opentelemetry/sdk/internal.rb#L51-L63

Reply:
Follow-up from SIG: This is fine, just document the duplication so that we know about this opportunity to refactor and can address it when we're ready to move the logs work into the main SDK.